api/v1/controllers/bots.js

```diff
@@ -150,7 +150,7 @@ module.exports = {
       seqObj.installedBy = user.id;
       u.setOwner(seqObj, req)
         .then(() => helper.model.create(seqObj))
-        .then((o) => o.reload())
+        .then((o) => o.reload(o._modelOptions.defaultScope))
```

Comment:
I only updated the cases which made tests to fail. Usually in most of the cases, we create an instance without the association, so it made sense as well that its mostly needed during updates. Is there any specific case you are looking for?
I would not be surprised if we have a use case where we need this change and we missed it because its not covered by tests.

Reply:
No specific case, just surprised I didn't see that form since we had discussed it.
I suppose it makes sense that the places we need to do reload are places where we are taking an existing model and updating it.

Reply:
Actually, looking through it again, there are several places we could do it that way instead:
Generator.createWithCollector
CollectorGroup.createCollectorGroup
postBots
postEvents
postLens
All the tests where the model is created in the setup

There's also a bunch of places where the instance comes from a find. I don't think this is necessary there, that object should already have the scope set. Did you try those with a plain reload?

Reply:
I think the only place it should be necessary to pass in the model to the reload is the instance methods. 
(Generator.updateWithCollectors, CollectorGroup.addCollectorsToGroup, CollectorGroup.patchCollectors)

Reply:
Well, if you are passing in associations while creating the object, then also we will need to pass in scope to reload to get back the associations.

Reply:
If you set the association when creating the object, the instance should be scoped and the reload will use that scope by default.

Reply:
> There's also a bunch of places where the instance comes from a find
Cant locate it, which places?. I wonder why we were doing the reload before in this case.

Reply:
patchCollectorGroup, for example.

Reply:
@jgraff2 made changes as suggested. 2 things to notice:
1. Some places, when I add the default scope when creating, it expects association object instead of just association name. For example,
In db/model/collectorgroup.js, we pass {name: 'cg', desc: 'desc', collectors: ['col1']} to create method and we add default Scope. When I run the test, I get this error:
//SequelizeValidationError: notNull Violation: Collector.name cannot be null,// notNull Violation: Collector.version cannot be null//

So, I wasnt able to get that working with create in those places

2. Some places instance comes from find, but we setOwner after that, and then reload. Does not work without passing defaultScope in reload because owner is not set otherwise.